lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -27,9 +27,118 @@ class AwsCompileCloudFrontEvents {
       'before:remove:remove': this.logRemoveReminder.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8025
+    const originObjectSchema = {
+      type: 'object',
+      properties: {
+        ConnectionAttempts: { type: 'integer', miminum: 1, maximum: 3 },
+        ConnectionTimeout: { type: 'integer', miminum: 1, maximum: 10 },
+        CustomOriginConfig: {
+          type: 'object',
+          properties: {
+            HTTPPort: { type: 'integer' },
+            HTTPSPort: { type: 'integer' },
+            OriginKeepaliveTimeout: { type: 'integer', miminum: 1, maximum: 60 },
+            OriginProtocolPolicy: {
+              type: 'string',
+              enum: ['http-only', 'match-viewer', 'https-only'],
+            },
+            OriginReadTimeout: { type: 'integer', miminum: 1, maximum: 60 },
+            OriginSSLProtocols: {
+              type: 'array',
+              items: { type: 'string', enum: ['SSLv3', 'TLSv1', 'TLSv1.1', 'TLSv1.2'] },
+            },
+          },
+          additionalProperties: false,
+          required: ['OriginProtocolPolicy'],
+        },
+        DomainName: { type: 'string' },
+        Id: { type: 'string' },
```

Comment:
I think we should not include `Id`, as we unconditionally generate it: https://github.com/serverless/serverless/blob/6f392b3305e545383b2b80775b1a9e6bf6e2d8b6/lib/plugins/aws/package/compile/events/cloudFront/index.js#L149